dnf/rpm/__init__.py

```diff
@@ -48,7 +48,11 @@ def detect_releasever(installroot):
             raise dnf.exceptions.Error(msg)
         releasever = hdr['version']
         try:
-            off = hdr[rpm.RPMTAG_PROVIDENAME].index(distroverpkg)
+            try:
+                # rpm has switched from bytes to strings -> this works only on old rpm versions
+                off = hdr[rpm.RPMTAG_PROVIDENAME].index(distroverpkg)
+            except ValueError:
+                off = hdr[rpm.RPMTAG_PROVIDENAME].index(distroverpkg.decode("utf8"))
             flag = hdr[rpm.RPMTAG_PROVIDEFLAGS][off]
             ver = hdr[rpm.RPMTAG_PROVIDEVERSION][off]
             if flag == rpm.RPMSENSE_EQUAL and ver:
```

Comment:
The very same problem remains on the line 55 (github doesn't allow me to comment directly on that line):
`if hdr['name'] != distroverpkg:`
distroverpkg is of bytes type, hdr['name'] is str, so the condition will always be true.

Reply:
Thanks for catching this. Fixed.